packages/s3-store/index.ts

```diff
@@ -498,7 +498,16 @@ export class S3Store extends DataStore {
 
     if (metadata.file.size === newOffset) {
       try {
-        const parts = await this.retrieveParts(id)
+        let parts = await this.retrieveParts(id)
+        // If no parts exist yet, then the incomplete part needs to be completed
+        if (!parts) {
+          const incompletePart = await this.getIncompletePart(id)
+          if (incompletePart) {
+            await this.uploadPart(metadata, incompletePart, nextPartNumber)
+            parts = await this.retrieveParts(id)
+          }
+        }
+
```

Comment:
Yep, this makes sense. I thought I was seeing successful outcomes when there was at least 1 part -- but now I need to double check because, logically, the same behavior should exist regardless when there's incomplete parts, regardless of existing parts 👍 

Will tweak this to:
```ts
const incompletePart = await this.getIncompletePart(id)
if (incompletePart) {
  await this.uploadPart(metadata, incompletePart, nextPartNumber)
}

const parts = await this.retrieveParts(id)
```